syft/generic/string.py

```diff
@@ -0,0 +1,173 @@
+        location: BaseWorker = None,
+        id_at_location: (str or int) = None,
+        register: bool = False,
+        owner: BaseWorker = None,
+        ptr_id: (str or int) = None,
+        garbage_collect_data: bool = True,
+    ):
+        """
+           Creates a StringPointer object that points to a String object 'obj'
+           after sending the latter to the worker 'location'.
+
+           Returns:
+                  a StringPointer object
+        """
+
+        if id_at_location is None:
+            id_at_location = obj.id
+
+        if owner is None:
+            owner = obj.owner
+
+        string_pointer = StringPointer(
+            location=location,
+            id_at_location=id_at_location,
+            owner=owner,
+            id=ptr_id,
+            garbage_collect_data=garbage_collect_data,
+        )
+
+        return string_pointer
+
+    @staticmethod
+    def simplify(worker: BaseWorker, string: "String"):
+        """
+           Breaks String object into a tuple of simpler objects, its constituting objects that are
+           serializable.
+
+
+           Args:
+
+               worker: a BaseWorker object
+               string: the String object to be simplified
+
+           Returns:
+               
+               A tuple of simpler objects that are sufficient to recreate
+               a String object that is a clone of `string`.
+
+        """
+
+        def simplify_str(str):
+            return str.encode("utf-8") if str else None
+
+        # Encode the string into a bytes object
+        simple_core = simplify_str(string.child)
+
+        tags = (
+            [simplify_str(tag) if isinstance(tag, str) else tag for tag in string.tags]
+            if string.tags
+            else None
+        )
+        description = simplify_str(string.description)
+
+        return (simple_core, string.id, tags, description)
+
+    @staticmethod
+    def detail(worker: BaseWorker, simple_obj: Tuple):
+        """
+           Create an object of type String from the reduced representation in `simple_obj`.
+
+
+           Args:
+
+               worker: BaseWorker
+                       The worker on which the new String object is to be created.
+               simple_obj: tuple
+                           A tuple resulting from the serialized then deserialized returned tuple
+                           from the `simplify` static method above.
+
+           Returns:
+
+               A String object
+
+
+        """
+
+        # Get the contents of the tuple represening the simplified object
+        simple_core, id, tags, description = simple_obj
+
+        def detail_bstr(b_str):
+            return str(b_str, encoding="utf-8") if b_str else None
+
+        # It appears that all strings are converted to bytes objects
+        # after deserialization, convert them back to strings
+        tags = (
+            [detail_bstr(tag) if isinstance(tag, bytes) else tag for tag in tags] if tags else None
+        )
```

Comment:
smae here
can't you do just `tags = sy.serde._detail(worker, string.tags)`?

Reply:
Done